sat-plugin/src/main/java/org/openhab/tools/analysis/report/ReportMojo.java

```diff
@@ -342,12 +364,21 @@ private int countPriority(NodeList messages, String priority) {
     }
 
     private void failOnErrors(File mergedReport) throws MojoFailureException {
-        int errorCount = selectNodes(mergedReport, "/sca/file/message[@priority=1]").getLength();
+        List<String> errorPriorities = new ArrayList<>();
+        addLevel(failOnError, "@priority=1", errorPriorities);
+        addLevel(failOnWarning, "@priority=2", errorPriorities);
+        addLevel(failOnDebug, "@priority=3", errorPriorities);
+        String xpathExpression = "/sca/file/message[" + String.join(" or ", errorPriorities) + "]";
+        int errorCount = selectNodes(mergedReport, xpathExpression).getLength();
         if (errorCount > 0) {
-            throw new MojoFailureException(String.format(
-                    "%n" + "Code Analysis Tool has found %d error(s)! %n"
-                            + "Please fix the errors and rerun the build. %n",
-                    selectNodes(mergedReport, "/sca/file/message[@priority=1]").getLength()));
+            throw new MojoFailureException(String.format("%n" + "Code Analysis Tool has found %d error(s)! %n"
+                    + "Please fix the errors and rerun the build. %n", errorCount));
```

Comment:
This message may need an update as well depending on the configuration. Users might think only fixing errors should be enough but if you configure it to fail on warnings/info this is no longer the case.

Reply:
Updated the message to detail the levels to fix.